pkg/storage/ingest/util.go

```diff
@@ -68,6 +69,7 @@ func commonKafkaClientOptions(cfg KafkaConfig, metrics *kprom.Metrics, logger lo
 		kgo.MetadataMaxAge(10 * time.Second),
 
 		kgo.WithLogger(newKafkaLogger(logger)),
+		kgo.WithHooks(kotel.NewKotel(kotel.WithTracer(kotel.NewTracer())).Hooks()...),
```

Comment:
Note that here we rely on OTel's global tracer, rather than trying to pass the tracer as a dependency. I believe, this is fine and this is how tracing is used in gRPC and HTTP clients/servers.